contracts/CharityChallenge.sol

```diff
@@ -89,8 +103,19 @@ contract CharityChallenge {
             isEventFinalizedAndValid = true;
             if (hasChallengeAccomplished) {
                 uint256 totalContractBalance = address(this).balance;
-                npoAddress.transfer(address(this).balance);
-                emit Donated(npoAddress, totalContractBalance);
+                uint8 sumRatio = 0;
+                uint length = npoAddresses.length;
+                for (uint i = 0; i < length; i++) {
+                  address payable npo = npoAddresses[i];
+                  sumRatio += npoRatios[npo];
+                }
+                for (uint i = 0; i < length; i++) {
+                  address payable npo = npoAddresses[i];
+                  uint8 ratio = npoRatios[npo];
+                  uint256 amount = totalContractBalance * ratio / sumRatio;
```

Comment:
this will have rounding error which results in some money left in contract. Solidity does not support decimal well

Reply:
I think we can control the ratio like [333 333 334] instead of [1 1 1]. wdyt?